### PR TITLE
add optional parameter `fromNote` to `note.build`

### PR DESCRIPTION
**Pull request by holden-caulfield**

### CODE REVIEW

**Comment by danigb** [comment]
Great! It looks good. I'm outside the weekend. I'll take a deep look on monday and merge. Thanks

File: packages/note/index.js [hunk 1]
```diff
@@ -277,15 +277,21 @@ export const altToAcc = alt =>
...
  * Note.build({ step: 5 }) // => "A"
  * Note.build({ step: 1, acc: -1 }) // => "Db"
  * Note.build({ step: 2, acc: 2, oct: 2 }) // => "E##2"
  * Note.build({ step: 7 }) // => null
+ * Note.build({alt: 1, oct: 3}, "C4") // => "C3"
  */
-export const build = ({ step, alt, oct } = {}) => {
+export const build = (buildProps = {}, fromNote = null) => {
+  const fromProps = fromNote ? props(fromNote) : {};
+  const { step, alt, oct } = Object.assign({}, fromProps, buildProps);
```
Comment (danigb) [inline packages/note/index.js]: I would write the two previous lines as following:

```js
const { step, alt, oct } = fromNote ? Object.assign({}, props(fromNote), buildProps) : buildProps;
```

What do you think?

**Comment by danigb** [comment]
I've added a comment to your PR. Tell me what do you think about it.

Also, with this functionality I think `build` is not a great name. But I think that `set` is also not valid since you can pass only an object. I would love to find a better name. Ideas?

File: packages/note/index.js (hunk 1 above)
Comment (holden-caulfield) [reply packages/note/index.js]: Yes! Makes total sense. I will make this change.

**Comment by holden-caulfield** [comment]
Let me think of a better name. Thanks for the feedback!

**Comment by holden-caulfield** [comment]
Hey @danigb made some changes on the PR. I've added your suggestion to make the code cleaner, and also tried renaming from `build` to `from`, which was the name that felt better to me (you are creating a note "from" a set of props, and optionally another note).

If that does not sound like a good name, just in case I was also thinking `withProps` or maybe `fromProps`.

I've kept `build` on the API as a deprecated method to avoid introducing a breaking change

**Comment by danigb** [comment]
:tada: Great! I like the new (old) name. Thanks for taking time to build this PR :+1: And +1 to the deprecation (instead of breaking change)

**Comment by holden-caulfield** [comment]
You're welcome. Happy to contribute on this amazing lib

**Pull request merged**